pkg/virt-api/rest/subresource.go

```diff
@@ -726,6 +655,37 @@ func (app *SubresourceAPIApp) fetchVirtualMachineInstance(name string, namespace
 	return vmi, nil
 }
 
+// FetchVirtualMachineInstanceForVM by namespace and name
+func (app *SubresourceAPIApp) FetchVirtualMachineInstanceForVM(namespace, name string) (*v1.VirtualMachineInstance, *errors.StatusError) {
+	vm, err := app.virtCli.VirtualMachine(namespace).Get(name, &k8smetav1.GetOptions{})
```

Comment:
I'm not 100% sure that we really need both GETs as it is a 1on1 mapping and the only difference would be the error we return. But I'm fine with it this way as it is clearer.
If I understand correctly, the GET should be a cache hit anyways.